### ISSUE TITLE
doesn't work and doesn't fail as described in documentation

### ISSUE THREAD

**Issue by glebovitz** [auth0/react-native-lock#127]
Thank you for using Lock! Please read the following before creating a new issue.

- [ ] Is this a question about Auth0 in general or a support request? Get in touch with us at https://support.auth0.com.
- [ ] Is this issue something you can fix? Please send us a pull request! Bug fixes are welcome.

Otherwise, create a new issue with the following information:
-------------------------------------------------------------

- [X ] Code snippet or sample project that reproduces the bug
I am using the example downloaded from the web. Project doesn't build under react-native 0.41.
- [X] Screenshots when appropriate
** BUILD FAILED **

The following build commands failed:
	CompileC /Users/gregg/Projects/src/classic/lock/ios/build/Build/Intermediates/A0RNLock.build/Debug-iphonesimulator/A0RNLock.build/Objects-normal/x86_64/A0LockReact.o A0RNLock/Core/A0LockReact.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
Installing build/Build/Products/Debug-iphonesimulator/Auth0Sample.app
An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=2):
Failed to install the requested application
An application bundle was not found at the provided path.
Provide a valid path to the desired application bundle.
Print: Entry, ":CFBundleIdentifier", Does Not Exist

Command failed: /usr/libexec/PlistBuddy -c Print:CFBundleIdentifier build/Build/Products/Debug-iphonesimulator/Auth0Sample.app/Info.plist
Print: Entry, ":CFBundleIdentifier", Does Not Exist

- [X ] Lock version
react-native-lock 0.4.0
- [X] iOS and/or Android version
IOS
- [X] If using CocoaPods, please include your Podfile
  # Uncomment this line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for Auth0Sample

  target 'Auth0SampleTests' do
    inherit! :search_paths
    # Pods for testing
  end

end

# Auth0 Lock

pod 'Lock', '~> 1.26'
pod 'Lock/TouchID'
pod 'Lock/SMS'
pod 'Lock/Email'
- [X] React Native & React version
├─┬ react@15.3.1
├─┬ react-native@0.33.1
│ │ ├── babel-plugin-transform-react-display-name@6.23.0
│ │ └─┬ babel-plugin-transform-react-jsx@6.23.0
│ │   └─┬ babel-helper-builder-react-jsx@6.23.0
│ ├─┬ babel-preset-react-native@1.9.1
│ │ ├─┬ babel-plugin-react-transform@2.0.2
│ │ ├── babel-plugin-transform-react-jsx-source@6.22.0
│ ├── react-clone-referenced-element@1.0.1
│ ├── react-timer-mixin@0.13.3
│ ├─┬ react-transform-hmr@1.0.4
│ │ └─┬ react-proxy@1.1.8
│ │   └── react-deep-force-update@1.0.1
├─┬ react-native-lock@0.4.0
│ └── react-native-auth0@0.0.1
- [ ] NPM version

Make sure to include **as much information as possible** for us to understand and reproduce the bug, that way we can fix it as quickly as possible.

**Comment by hzalaz** [comment]
Using the fix proposed in https://github.com/auth0/react-native-lock/issues/123#issuecomment-281478242 I got this working. The error is not helpful but maybe you forgot to run `pod install` inside the `iOS` folder

**Comment by glebovitz** [comment]
I will give that a try. I followed the instructions on the site and didn’t see anything about running pod install. If this works then it might be useful to update the site.

By the way, I did run rnpm link react-native-lock which should run a pod install.

Gregg

**Comment by hzalaz** [comment]
Yes but it fails like #123 says so you need to patch like I described and re-run the command

**Comment by glebovitz** [comment]
Okay. Got it working using the instructions on the quickstart page with the added step of removing the tvOS and test targets.

Seems to work so far. Will do more testing tonight.

Thanks for your help.

**Comment by glebovitz** [comment]
Hi Hernan,

I am a bit stuck and not moving forward with Auth0. I cannot find answers to my questions on the forums (the answers provided don’t seem to work with the auth0 client).

Here is the synopsis of my issue:

I want to keep all my user profile information in Auth0, but keep application specific preferences in my application server. The only relationship between the user and server is the JWT provided by the client. The JWT by default keeps authentication service specific identification information (google id for example is the authentication is done via google).

I want my JWT to have a identity that is assigned by auth0 that is independent of the identity provider. This will enable multiple identities for each user in the future.

Is there a way to get a auth0 universal user identity into the JWT provided by the react-native-lock client? If not, is there a way to do this using hooks?

Thank you for your help,

Gregg


Gregg Lebovitz
gregg@lebovitz.net
+1.617.396-4114 Phone
+1.617.233.5504 Mobile
+1.617.505.1232 Fax
www.lebovitz.net
